scripts/hud/dfjump.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const COLOR_CLASS = {
+	AIR: 'dfjump__press--air',
+	GROUND: 'dfjump__press--ground'
+};
+
+class DFJump {
+	/** @type {Panel} @static */
+	static container = $('#DFJumpContainer');
+	/** @type {ProgressBar} @static */
+	static releaseBar = $('#JumpReleaseBar');
+	/** @type {ProgressBar} @static */
+	static pressBar = $('#JumpPressBar');
+	/** @type {Label} @static */
+	static releaseLabel = $('#JumpReleaseLabel');
+	/** @type {Label} @static */
+	static pressLabel = $('#JumpPressLabel');
+	/** @type {Label} @static */
+	static totalLabel = $('#JumpTotalLabel');
+
+	static onLoad() {
+		this.initializeSettings();
+		this.colorClass = COLOR_CLASS.GROUND;
+	}
+
+	static onDFJumpUpdate(releaseDelay, pressDelay, totalDelay) {
+		const releaseRatio = releaseDelay * this.inverseMaxDelay;
+		const pressRatio = Math.abs(pressDelay) * this.inverseMaxDelay;
+		const newPressColorClass = pressDelay < 0 ? COLOR_CLASS.GROUND : COLOR_CLASS.AIR;
+
+		this.releaseBar.value = releaseRatio;
+		this.pressBar.value = pressRatio;
+		this.pressBar.RemoveClass(this.colorClass);
+		this.pressBar.AddClass(newPressColorClass);
+		this.colorClass = newPressColorClass;
+
+		this.releaseLabel.text = releaseDelay.toFixed();
+		this.pressLabel.text = pressDelay.toFixed();
+		this.totalLabel.text = totalDelay.toFixed();
+	}
+
+	static setMaxDelay(newDelay) {
+		this.maxDelay = this.nullCheck(newDelay, 1);
+		this.inverseMaxDelay = 1.0 / this.maxDelay;
+	}
+
+	static nullCheck(val, def) {
+		return val === null ? def : val;
+	}
```

Comment:
Welcome to the world of JavaScript, where you can simply do `return val ?? def` here, the "nullish coalescing" operator: https://sebhastian.com/javascript-double-question-mark/

In fact, the function basically becomes unnecessary then and all invokes can be replaced with the double question mark.

You will want to test this, I believe our V8 version is new enough to handle it.

Reply:
wow. This is really nice for this use case.

I'll remove the `nullCheck()` method.